rust/auction/program/src/processor.rs

```diff
@@ -559,66 +559,86 @@ impl BidState {
 
         match self {
             // In a capped auction, track the limited number of winners.
-            BidState::EnglishAuction { ref mut bids, max } => {
-                match bids.last() {
-                    Some(top) => {
-                        msg!("Looking to go over the loop, but check tick size first");
-
-                        for i in (0..bids.len()).rev() {
-                            msg!("Comparison of {:?} and {:?} for {:?}", bids[i].1, bid.1, i);
-                            if bids[i].1 < bid.1 {
-                                if let Some(gap_tick) = gap_tick_size_percentage {
-                                    BidState::assert_valid_gap_insertion(gap_tick, &bids[i], &bid)?
-                                }
-
-                                msg!("Ok we can do an insert");
-                                if i + 1 < bids.len() {
-                                    msg!("Doing a normal insert");
-                                    bids.insert(i + 1, bid);
-                                } else {
-                                    msg!("Doing an on the end insert");
-                                    bids.push(bid)
-                                }
-                                break;
-                            } else if bids[i].1 == bid.1 {
-                                if let Some(gap_tick) = gap_tick_size_percentage {
-                                    if gap_tick > 0 {
-                                        msg!("Rejecting same-bid insert due to gap tick size of {:?}", gap_tick);
-                                        return Err(AuctionError::GapBetweenBidsTooSmall.into());
+            BidState::EnglishAuction {
+                ref mut bids,
+                ref mut max,
+            } => {
+                // check for instance sale auction
+                if instant_sale_price.is_some() {
+                    if *max > 0 {
+                        max.checked_sub(1)
+                            .ok_or(AuctionError::NumericalOverflowError)?;
+                        *max = *max - 1;
```

Comment:
Here should be `*max = max.checked_sub(1).ok_or(AuctionError::NumericalOverflowError)?;`

Reply:
fixed